locale.go

```diff
@@ -109,9 +112,12 @@ func (service *LocalesService) Upsert(ctx context.Context, spaceID string, local
 	var path string
 	var method string
 
-	if locale.Sys != nil && locale.Sys.CreatedAt != "" {
+	if locale.Sys != nil && locale.Sys.CreatedAt != ""  {
 		path = fmt.Sprintf("/spaces/%s/locales/%s", spaceID, locale.Sys.ID)
```

Comment:
I think `locale` always needs environments. https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/locales

So, Could you change it? And you can use `service.c.Environment` instead of adding new field.

```suggestion
		path = fmt.Sprintf("/spaces/%s/environments/%s/locales/%s", spaceID, service.c.Environment, locale.Sys.ID)
```

Reply:
that looks like the Content Delivery API, are we using the CMA instead? https://www.contentful.com/developers/docs/references/content-management-api/#/reference/locales/locale-collection/create-a-locale/console

Reply:
My apologies. I made a mistake with the link. And you can change the path as above. Because the CMA reference shows the path as `https://api.contentful.com/spaces/{space_id}/environments/{environment_id}/locales`.

Reply:
> Locales are environment-level entities.

https://www.contentful.com/developers/docs/tutorials/general/setting-locales/

I assume that at the time this code was written, locale was a space-level entity, not per-environment. (for some reason it still works today).

For example, [this document](https://www.contentful.com/developers/docs/tutorials/general/setting-locales/) has space-level locale request.

```
curl -X POST
  -H "Authorization: Bearer <CONTENT_MANAGEMENT_KEY>"
  -H "Content-Type: application/vnd.contentful.management.v1+json"
  -d '{
   "name":"English (British)",
   "code":"en-GB"
}'
"https://api.contentful.com/spaces/<SPACE_ID>/locales"
```

But, in API reference, there aren't the above request. So I want to change all request from space-level locale to entity-level locale.

So, could you try like the following change to the all locale path instead of adding `ENV_ID` field?

```diff
-		path = fmt.Sprintf("/spaces/%s/locales/%s", spaceID, locale.Sys.ID)
+		path = fmt.Sprintf("/spaces/%s/environments/%s/locales/%s", spaceID, service.c.Environment, locale.Sys.ID)
```